bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml

```diff
@@ -73,4 +73,12 @@
 		</config-description>
 	</thing-type>
 
+	<thing-type id="delayed-thing">
+        <label>Magic Delayed Online Thing</label>
+        <description>A thing that goes online after some time.</description>
+
+        <channels>
+            <channel id="switch" typeId="switch" />
```

Comment:
You could use a number and the implementation could use that number of seconds to delay the go online state update. A negative number could schedule a go offline state update. (Just an idea)

Reply:
Yes, nice. And for a new command to this channel the Thing will go OFFLINE, wait the amount of seconds from the command (if positive) and go ONLINE again.

Reply:
@htreu Okay for me, too. But my suggestion has been a different one :wink: 
My suggestion has been:
* Receiving a command does not change the state immediately.
* If the received number is > 0, then an online state update will be scheduled with a delay of number.
* If the received number is < 0, then an offline state update will be scheduled with a delay of -number.

So, in short:
* delay = Math.abs(number)
* state = number > 0 ? ONLINE : OFFLINE

Reply:
Good idea, I have changed the code to reflect that and as a bonus I can reproduce #4931 with that!